README.md

```diff
@@ -123,3 +153,60 @@ cd ./api/
 # Run 'npm install' instead if you are using npm
 yarn
 ```
+
+## Starting two stacks simultaneously
+
+It might be necessary to create two Docker stacks for two verions of CodePod, respectively. For example, you might want to test the new version of CodePod while keeping the old version running.
+
+Because Docker uses the folder name as the default suffix in container names, these two stacks may conflict with each other. To avoid this, you can use the `COMPOSE_PROJECT_NAME` environment variable to set a prefix for the container names. For example, you can set `COMPOSE_PROJECT_NAME=codepod-v2` in the `CODEPOD_ROOT/compose/dev/.env` file of the new stack, and then [start](#starting-the-stack) the new stack. 
+
+The two stacks also may share the same network ports due to the same configuration files used. To set the ports, search for the following lines in  `CODEPOD_ROOT/compose/dev/nginx.conf` (two occurences of the two lines in the file) file of the new stack:
+
+```yaml
+        listen 80; 
+        listen [::]:80;
+```
+
+and the following section in the `CODEPOD_ROOT/compose/dev/compose.yml` file of the new stack:
+
+```
+  nginx:
+    image: nginx:alpine
+    ports:
+      - 80:80
+    volumes:
+      - ./nginx.conf:/etc/nginx/conf.d/default.conf
+```
+
+and replace the default port number 80 to a new port number. For example, you can set the port number to 8080 for all occurences of 80. 
+
+Also, comment out the port section of the `ui` container in `CODEPOD_ROOT/compose/dev/compose.yml` as: 
```

Comment:
Let's mention the "development process" somewhere. Basically, the process is:

1. start the docker stack
2. modify some `.ts` files. Upon saving, the server processes will be restarted automatically.
    - How it works:
       - the source files are mounted into the `ui/` or `api/` container in the development stack.
       - the server process monitor the file changes and restart accordingly
4. see the changed behavior in app by refreshing the webpage.

Reply:
In the future, I plan to remove the "hot-reloading" feature of React (by not binding `ui:3000` to `localhost:3000`) because hot-reloading seems to only update the app partially. A manual refresh is safer and more reliable. Hot reloading is reliable for small apps or style changes, but not a complex app like CodePod. But let's keep it as is for now.